catalog/solr/catalog-solr-offline-gazetteer/src/test/groovy/ddf/catalog/solr/offlinegazetteer/GazetteerQueryOfflineSolrSpec.groovy

```diff
@@ -0,0 +1,344 @@
+                getSuggesterResponse() >> Mock(SuggesterResponse) {
+                    getSuggestions() >>
+                            [(GazetteerQueryOfflineSolr.SUGGEST_PLACE_KEY): [Mock(Suggestion) {
+                                getPayload() >> "id"
+                                getTerm() >> "title"
+
+                            }]]
+                }
+            }
+        }
+        when:
+        List<Suggestion> results = testedClass.getSuggestedNames("place", maxResults)
+
+        then:
+        results.size() == 1
+    }
+
+    def "test getSuggestedNames solrclient exception"() {
+        setup:
+        int maxResults = 10
+        1 * solrClient.query(*_) >> { throw new SolrServerException("exception") }
+
+        when:
+        List<Suggestion> results = testedClass.getSuggestedNames("place", maxResults)
+
+        then:
+        GeoEntryQueryException e = thrown()
+    }
+
+    def "getNearestCities"() {
+        setup:
+        String locationWkt = "POINT (-98.86253 29.18968)"
+        String pointAbout42kmAway = "POINT (-98.496708 29.418376)"
+        int radiusInKm = 50
+        int maxResults = 10
+        1 * solrClient.query(*_) >> { SolrQuery query, METHOD method ->
+            assert METHOD.POST == method
+
+
+            // TODO (RCZ) - I really don't like extracting the polygon from the solr query. its
+            // risky logic and closely coupling to the implementation. But I also don't want to
+            // open access and have a @VisibleForTesting..
+
+            // A point about 42km away should fall well within searching for cities within 50miles
+            WKTReader reader = new WKTReader()
+            String WKTPolygon = extractIntersectsPolygon(query.getQuery())
+            assert reader.read(WKTPolygon).contains(reader.read(pointAbout42kmAway))
+
+            Mock(QueryResponse) {
+                getResults() >> Mock(SolrDocumentList) {
+                    stream() >> {
+                        Stream.of(Mock(SolrDocument) {
+                            get("title_txt") >> ["title"]
+                            get("location_geo") >> [pointAbout42kmAway]
+                        })
+                    }
+                }
+            }
+        }
+
+        when:
+        List<NearbyLocation> results = testedClass.
+                getNearestCities(locationWkt, radiusInKm, maxResults)
+
+        then:
+        results.size() == 1
+        with(results.first()) {
+            cardinalDirection == "NE"
+            40 <= it.distance && it.distance <= 45
+            it.name == "title"
+        }
+    }
+
+    String extractIntersectsPolygon(String query) {
+        String[] arr = query.split(":", 2)
+        assert arr.length == 2
+        List<String> queryNodes = Arrays.asList(
+                arr[1].
+                        replace("\"", "").
+                        split("AND|OR"))
+        String intersectsQuery = queryNodes.find { it.contains("POLYGON") }
```

Comment:
@rzwiefel this should probably look for contains 'Intersects' (or intersects and polygon) since we do the replace on polygon... that is if we keep this logic at all.